### ISSUE TITLE
Negative KL divergence

### ISSUE THREAD

**Issue by arnauqb** [VincentStimper/normalizing-flows#29]
Hello, first of all, thanks for the great package! I am noticing that for the NF given in the README, I obtain negative KL values when training, but this does not happen with other flows. I have realized that if I set the `scale_map` parameter to `sigmoid` the problem is gone. Here is an example where I try to fit a Normal distribution:

```python
import normflows as nf
import torch

def make_flow(scale_map="sigmoid"):
    base = nf.distributions.base.DiagGaussian(4)
    num_layers = 5
    flows = []
    for i in range(num_layers):
        param_map = nf.nets.MLP([2, 64, 2], init_zeros=True)
        flows.append(nf.flows.AffineCouplingBlock(param_map, scale_map=scale_map))
        flows.append(nf.flows.Permute(4, mode='shuffle'))
    model = nf.NormalizingFlow(base, flows)
    return model

def train_flow(flow, target):
    optimizer = torch.optim.Adam(flow.parameters(), lr=5e-3)
    best_loss = torch.inf
    while True:
        n_epochs_without_improvement = 0
        optimizer.zero_grad()
        flow_samples, flow_lps = flow.sample(10000)
        target_lps = target.log_prob(flow_samples)
        loss = (flow_lps - target_lps).mean()
        if loss < best_loss:
            best_loss = loss.item()
            print(best_loss)
            n_epochs_without_improvementchs_without_improvement = 0
        else:
            n_epochs_without_improvement += 1
        if n_epochs_without_improvement > 50:
            break
        loss.backward()
        optimizer.step()
    return best_loss

target = torch.distributions.MultivariateNormal(torch.tensor([-3, -2., -1., 0.]), torch.eye(4))

flow_sigmoid = make_flow(scale_map = "sigmoid")
best_loss_sigmoid = train_flow(flow_sigmoid, target)
# This returns a value close to 0, as you would expect.

flow_exp = make_flow(scale_map = "exp")
best_loss_exp = train_flow(flow_exp, target)
# This returns a very negative value.
```

My first thought was numerical error but I think the value is too negative to be the case. Any help?

Thank you very much!

**Comment by VincentStimper** [comment]
Hi @arnauqb,

the `param_map` needs to return two parameters per transformed dimension, i.e. a shift and a scale. However, you have to dimensions being transformed, but only return two parameters and not four. This will not result in an error because the shift and scale is broadcasted to all dimensions, but the computation of the log determinant and, hence, the log likelihood will be inconsistent. So, please try again with
```
param_map = nf.nets.MLP([2, 64, 4], init_zeros=True)
```
I think this should resolve the issue. 

Best regards,
Vincent

**Comment by arnauqb** [comment]
Oh I see! Thank you for your prompt answer, this was causing me a headache :)
